cmd/sftp-server-driver.go

```diff
@@ -201,19 +201,83 @@ func (f *sftpDriver) Fileread(r *sftp.Request) (ra io.ReaderAt, err error) {
 	return obj, nil
 }
 
+func (w *writerAt) Close() error {
+	for i := range w.buffers {
```

Comment:
Check if w.remaining > 0 - return error if so - and use w.w.CloseWithError.

Reply:
will do